applications/StructuralMechanicsApplication/custom_elements/total_lagrangian.cpp

```diff
@@ -235,17 +235,17 @@ namespace Kratos
     {
         KRATOS_TRY
         
-        const unsigned int number_of_nodes = GetGeometry().PointsNumber();
-        const unsigned int dimension = GetGeometry().WorkingSpaceDimension();
+        const unsigned int number_of_nodes = this->GetGeometry().PointsNumber();
+        const unsigned int dimension = this->GetGeometry().WorkingSpaceDimension();
 
         // For axisymmetric case
         Vector N;
         double Radius = 0.0f;
         
         if ( StrainSize == 4 )
         {
-            N = GetGeometry().ShapeFunctionsValues( N, IntegrationPoints[PointNumber].Coordinates() );
-            Radius = StructuralMechanicsMathUtilities::CalculateRadius(N, GetGeometry());
+            N = this->GetGeometry().ShapeFunctionsValues( N, IntegrationPoints[PointNumber].Coordinates() );
```

Comment:
Same

Reply:
The this-> is slow?

Reply:
No, what is slow is the call

ShapeFunctionsValues(vector, coordinates)

You should use instead the one that returns at once the one on all the integration points, smug like

ShapeFunctionsValues(matrix, integrstiontype)

Or eventually

ShapeFunctionsValues(vector, index)